### PR TITLE
treewide: fix luacheck warnings

### PR DESCRIPTION
**Pull request by bobcanthelpyou**

### CODE REVIEW

File: package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua [hunk 1]
```diff
@@ -97,7 +97,7 @@ end
 function Node:resolve_depends()
 	local updated = false
 	for _, node in ipairs(self.children) do
-		update = updated or node:resolve_depends()
+		updated = updated or node:resolve_depends()
```
Comment (NeoRaider) [inline package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua]: This one is an actual bugfix that changes logic, let's move that to a separate commit (can still be the same PR).

File: package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua [hunk 2]
```diff
@@ -257,7 +257,7 @@ function AbstractValue:handle()
 	end
 end
 
-function AbstractValue:write(value)
+function AbstractValue:write(_)
```
Comment (NeoRaider) [inline package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua]: Hmm, as this is supposed to be overridden, let's at least add a comment above. But also drop the unused `_` for consistency.

File: package/gluon-web/luasrc/usr/lib/lua/gluon/web/i18n.lua [hunk 3]
```diff
@@ -13,7 +13,7 @@ return function(config)
 		return string.format('%s/%s.%s.lmo', i18ndir, pkg, lang)
 	end
 
-	local function no_translation(key)
+	local function no_translation(_)
```
Comment (NeoRaider) [inline package/gluon-web/luasrc/usr/lib/lua/gluon/web/i18n.lua]: Drop the unused `_`.

File: package/gluon-config-mode-geo-location/luasrc/lib/gluon/config-mode/wizard/0400-geo-location.lua [hunk 4]
```diff
@@ -43,7 +43,7 @@ return function(form, uci)
 		end
 	end
 
-	local map = {}
+	local map
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-config-mode-geo-location/luasrc/lib/gluon/config-mode/wizard/0400-geo-location.lua]: Is `map.lon` and `map.lat` (or `map` at all) used somewhere (or how)?

[the remark this answers is not in the document]
Comment (NeoRaider) [reply package/gluon-config-mode-geo-location/luasrc/lib/gluon/config-mode/wizard/0400-geo-location.lua]: Yes, these are configuration values of the osm.MapValue.

File: package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless [hunk 5]
```diff
@@ -16,22 +16,20 @@ if not sysconfig.gluon_version then
 	-- capable of operating in the 2.4 GHz and 5 GHz band need
 	-- to be distributed evenly.
 	local radio_band_count = {band24=0, band5=0}
-	util.foreach_radio(uci, function(radio, index, config)
+	util.foreach_radio(uci, function(radio, _, _)
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless]: Should be removed

File: package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless [hunk 6]
```diff
@@ -16,22 +16,20 @@ if not sysconfig.gluon_version then
...
+		elseif (hwmodes.a or hwmodes.ac) and not hwmodes.g then
 			-- 5 GHz
 			radio_band_count["band5"] = radio_band_count["band5"] + 1
 		end
 	end)
 
 	-- Use the number of all fixed 2.4G GHz and 5 GHz radios to
 	-- distribute dualband radios in this step.
-	util.foreach_radio(uci, function(radio, index, config)
+	util.foreach_radio(uci, function(radio, _, _)
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless]: Should be removed

File: package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua [hunk 7]
```diff
@@ -63,23 +62,23 @@ local function action_upgrade(http, renderer)
...
+	local function image_supported(supported_tmpfile)
+		return (os.execute(string.format("exec /sbin/sysupgrade -T %q >/dev/null", supported_tmpfile)) == 0)
 	end
 
 	local function storage_size()
 		local size = 0
 		if unistd.access("/proc/mtd") then
 			for l in io.lines("/proc/mtd") do
-				local d, s, e, n = l:match('^([^%s]+)%s+([^%s]+)%s+([^%s]+)%s+"([^%s]+)"')
+				local _, s, _, n = l:match('^([^%s]+)%s+([^%s]+)%s+([^%s]+)%s+"([^%s]+)"')
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua]: Probably removed the unused groups, could you check again?

File: package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua [hunk 8]
```diff
@@ -63,23 +62,23 @@ local function action_upgrade(http, renderer)
...
 				if n == "linux" then
 					size = tonumber(s, 16)
 					break
 				end
 			end
 		elseif unistd.access("/proc/partitions") then
 			for l in io.lines("/proc/partitions") do
-				local x, y, b, n = l:match('^%s*(%d+)%s+(%d+)%s+([^%s]+)%s+([^%s]+)')
+				local _, _, b, n = l:match('^%s*(%d+)%s+(%d+)%s+([^%s]+)%s+([^%s]+)')
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua]: Probably removed the unused groups, could you check again?

File: package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua (hunk 1 above)
Comment (bobcanthelpyou) [reply package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua]: It is move to a separate commit

File: package/gluon-web/luasrc/usr/lib/lua/gluon/web/i18n.lua (hunk 3 above)
Comment (bobcanthelpyou) [reply package/gluon-web/luasrc/usr/lib/lua/gluon/web/i18n.lua]: Dropped

File: package/gluon-config-mode-geo-location/luasrc/lib/gluon/config-mode/wizard/0400-geo-location.lua (hunk 4 above)
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-config-mode-geo-location/luasrc/lib/gluon/config-mode/wizard/0400-geo-location.lua]: Ah okay, please check again.

File: package/gluon-core/luasrc/usr/lib/lua/gluon/iputil.lua [hunk 9]
```diff
@@ -85,10 +85,9 @@ function M.mac_to_ip(prefix, mac)
 	local h3 = 0x100 * m5 + tonumber(m6, 16)
 	local h4 = 0x100 * tonumber(m7, 16) + tonumber(m8, 16)
 
-	local prefix, plen = string.match(prefix, '(.*)/(%d+)')
-	plen = tonumber(plen, 10)
+	prefix = string.match(prefix, '(.*)/%d+')
 
-	local p1, p2, p3, p4, p5, p6, p7, p8 = M.IPv6(prefix)
+	local p1, p2, p3, p4, _, _, _, _ = M.IPv6(prefix)
```
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-core/luasrc/usr/lib/lua/gluon/iputil.lua]: Dropped

File: package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua (hunk 2 above)
Comment (bobcanthelpyou) [reply package/gluon-web-model/luasrc/usr/lib/lua/gluon/web/model/classes.lua]: Added a comment and dropped `_`

File: package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless (hunk 5 above)
[the remark this answers is not in the document]
Comment (NeoRaider) [reply package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless]: This one still has trailing `, _, _`

File: package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua [hunk 10]
```diff
@@ -157,7 +157,7 @@ function M.find_phy(config)
 	end
 end
 
-local function get_addresses(uci, radio)
+local function get_addresses(_, radio)
```
Comment (NeoRaider) [inline package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua]: As this is a local function, we can also just remove the first argument and adjust the call sites.

File: scripts/check_site.lua [hunk 11]
```diff
@@ -37,7 +37,7 @@ local site, domain_code, domain, conf
 local function merge(a, b)
 	local function is_array(t)
 		local n = 0
-		for k, v in pairs(t) do
+		for _, _ in pairs(t) do
```
Comment (NeoRaider) [inline scripts/check_site.lua]: One `_` is enough.

File: package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless (hunk 5 above)
[the remark this answers is not in the document]
Comment (bobcanthelpyou) [reply package/gluon-core/luasrc/lib/gluon/upgrade/200-wireless]: Sorry missed it.

File: scripts/check_site.lua (hunk 11 above)
Comment (bobcanthelpyou) [reply scripts/check_site.lua]: Removed one of `_`

File: package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua (hunk 10 above)
Comment (bobcanthelpyou) [reply package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua]: The `_` "moved up" to `function M.get_wlan_mac(_, radio, index, vif)`

File: package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua [hunk 12]
```diff
@@ -18,7 +18,7 @@ local unistd = require 'posix.unistd'
 local tmpfile = "/tmp/firmware.img"
 
 
-local function filehandler(meta, chunk, eof)
+local function filehandler(_, chunk, eof)
```
Comment (NeoRaider) [inline package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua]: While the variable *file* is used across multiple calls of *filehandler*, it can at least be declared local to this file.

**Comment by NeoRaider** [comment]
Please also replace the stray `check_domain_switch` in `package/gluon-scheduled-domain-switch/check_site.lua` with `nil` to fix the last warning in the `package` dir.

File: package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua (hunk 12 above)
Comment (bobcanthelpyou) [reply package/gluon-web-admin/luasrc/lib/gluon/config-mode/controller/admin/upgrade.lua]: Add `local file` at the beginning

**Comment by bobcanthelpyou** [comment]
Thought the `check_domain_switch` will be fixed via different pull request, but replaced it.

**Comment by NeoRaider** [comment]
Ok, looking good. Please squash the fixup commits, then we can merge this.

**Comment by bobcanthelpyou** [comment]
Squashed
